src/main.rs

```diff
@@ -63,18 +63,23 @@ impl Snare {
     /// config file. **Note that because snare has multiple threads, the config file can change at
     /// any arbitrary point, not just after calling this function.**
     fn check_for_hup(&self) {
-        if self.sighup_occurred.load(Ordering::Release) {
-            let config = Config::from_path(&self.conf_path);
-            *self.config.lock().unwrap() = config;
-            self.sighup_occurred.store(false, Ordering::Release);
+        if self.sighup_occurred.load(Ordering::Relaxed) {
+            match Config::from_path(&self.conf_path) {
+                Ok(config) => *self.config.lock().unwrap() = config,
+                Err(msg) => eprintln!("{}", msg),
```

Comment:
So the error is printed on stderr, which may be in the background and may go un-noticed.

Not sure how you can fix that though.

One potential idea would be to have a `snare --reload` which communicates with the existing snare instance and prints any errors to its stderr, (not the daemon's). However, this would probably need more complex IPC.

Reply:
A later PR will send this to syslog.

Reply:
sounds good!